### DISCUSSION TITLE
Errors during createResource fetches

### DISCUSSION THREAD

**Discussion by yaq-cc** [solidjs/solid#1888] [Q&A]
### Describe the bug

Hi there,

I'm really struggling with errors coming from createResource.  No matter what I do, errors seem to go uncaught.  The resource state is updated to 'errored' but no errors appear in the actual error... I'm not quite sure this is the intended behavior.  

What am I doing wrong?

https://playground.solidjs.com/anonymous/ee5b8061-1ca0-4842-be13-21249d46f94e
```
import { render } from "solid-js/web";
import { createResource } from "solid-js";

function Joke() {
  const fetcher = fetchJoke({success: false, timeout: 1000});
  const [joke] = createResource(fetcher);
  return (<div>
		<p>Joke: {joke()}</p>
		<p>Loading: {joke.loading}</p>
		<p>Error: {joke.error}</p>
		<p>State: {joke.state}</p>
	</div>)
}

const fetchJoke = (params: { success: boolean, timeout: number}) => {
	return () => new Promise<string>((resolve, reject) => {
		setTimeout(() => {
			if (params.success) {
				resolve('a joke')
			} else {
				reject('no joke')
			}
		}, params.timeout)
	})
}

render(() => <Joke />, document.getElementById("app")!);
```

### Your Example Website or App

https://playground.solidjs.com/anonymous/ee5b8061-1ca0-4842-be13-21249d46f94e

### Steps to Reproduce the Bug or Issue

Check out https://playground.solidjs.com/anonymous/ee5b8061-1ca0-4842-be13-21249d46f94e

### Expected behavior

As a user, I expected createResource's error property to be filled along with the resource state set to errored.

### Platform

linux, macos, windows
"solid-js": "^1.7.6"

**Comment by ryansolid** [comment]
Sorry for the confusion. The state and like `.error` were added after the fact by request but sort of confuse how resources work primarily. That is when you access the value if it is error it throws up to the nearest ErrorBoundary. We error (and suspend) on read. Now it is possible to not use ErrorBoundary by guarding the value and only accessing `joke()` when there isn't an error, by using a Switch or some sort of conditional logic. This is also true of `.loading` to avoid a parent Suspense boundary from triggering.

For example, I imagine this will do what you expect:
```js
<div>
  <p>Joke: {!joke.loading && !joke.error && joke()}</p>
  <p>Loading: {joke.loading}</p>
  <p>Error: {joke.error}</p>
  <p>State: {joke.state}</p>
</div>
```

**Comment by emdede** [comment]
I had toyed around with your playground example but couldn't get reliable results. With Ryan's "hack" it now works reliably.

One thing that's missing is that `error` is an error instance and `loading` is a boolean. Afaict, the DOM insertion breaks here because you're not passing strings.

Here is an updated version of your example:
https://playground.solidjs.com/anonymous/64aaf0b4-b237-4a6f-b70c-698f1a1ca1f3

**Comment by yaq-cc** [comment]
Just chiming in here to say thank you for explaining everything here!

Regards,
